data_handler.py

```diff
@@ -0,0 +1,147 @@
+from auxiliaries import datetime_from_epoch
+from sender import ServerSender
+from sensor import Sensor
+from auxiliaries import set_verbosity
+from globalvalues import ANSI_RESET, ANSI_YEL, ANSI_GR, ANSI_RED
+from globalvalues import DEFAULT_DATA_BACKLOG_FILE
+from collections import deque
+import socket
+import time
+import ast
+import os
+
+CPM_DISPLAY_TEXT = (
+    '{{time}}: {yellow} {{counts}} cts{reset}' +
+    ' --- {green}{{cpm:.2f}} +/- {{cpm_err:.2f}} cpm{reset}' +
+    ' ({{start_time}} to {{end_time}})').format(
+    yellow=ANSI_YEL, reset=ANSI_RESET, green=ANSI_GR)
+strf = '%H:%M:%S'
+
+class Data_Handler(object):
+    """
+    Object for sending data to server. 
+
+    Also handles writing to datalog and 
+    storing to memory. 
+    """
+
+    def __init__(self,
+        	 manager=None,
+		 verbosity=1,
+	         logfile=None
+	         ):
+
+        self.v = verbosity
+	if manager and logfile is None:
+	    set_verbosity(self, logfile=manager.logfile)
+	else:
+	    set_verbosity(self, logfile=logfile)
+		
+	self.manager = manager
+    	
+    	self.queue = deque('')
+    	
+    def test_send(self, cpm, cpm_err):
+        """
+	Test Mode
+	"""
+	self.vprint(
+	    1, ANSI_RED + " * Test mode, not sending to server * " +
+	    ANSI_RESET)
+
+    def no_config_send(self, cpm, cpm_err):
+	"""
+	Configuration file not present
+	"""
+	self.vprint(1, "Missing config file, not sending to server")
+
+    def no_publickey_send(self, cpm, cpm_err):
+    	"""
+	Publickey not present
+	"""
+	self.vprint(1, "Missing public key, not sending to server")
+
+    def no_network_send(self, cpm, cpm_err):
+        """
+	Network is not up
+	"""
+	if self.manager.protocol == 'new':
+	    self.send_to_queue(cpm, cpm_err)
+	    self.vprint(1, "Network down, saving to queue in memory")
+	else:
+	    self.vprint(1, "Network down, not sending to server")  
+
+    def regular_send(self, cpm, cpm_err):
+        """
+	Normal send
+	"""
+	try:
+	    self.manager.sender.send_cpm(cpm, cpm_err)
+	    if self.manager.protocol == 'new':
+	        while self.queue:

```

Comment:
I know I had you remove `if self.queue` earlier, but it would be good to print a single message if there is data in the queue to flush to the server. So do an `if`, vprint(1, "Flushing memory queue to server"), then the while loop.